Sming/Libraries/Adafruit_GFX/BMPDraw.h

```diff
@@ -0,0 +1,193 @@
+/***************************************************
+  This is a library for the Adafruit 1.8" SPI display.
+
+This library works with the Adafruit 1.8" TFT Breakout w/SD card
+  ----> http://www.adafruit.com/products/358
+The 1.8" TFT shield
+  ----> https://www.adafruit.com/product/802
+The 1.44" TFT breakout
+  ----> https://www.adafruit.com/product/2088
+as well as Adafruit raw 1.8" TFT display
+  ----> http://www.adafruit.com/products/618
+
+  Check out the links above for our tutorials and wiring diagrams
+  These displays use SPI to communicate, 4 or 5 pins are required to
+  interface (RST is optional)
+  Adafruit invests time and resources providing this open source code,
+  please support Adafruit and open-source hardware by purchasing
+  products from Adafruit!
+
+  Written by Limor Fried/Ladyada for Adafruit Industries.
+  MIT license, all text above must be included in any redistribution
+ ****************************************************/
+/****************************************************
+ * ported for Sming by H.Boettcher.
+ * this implementation uses SPIFFS storage instead of SD cards
+ * hbottc@gmail.com
+ ***************************************************/
+/*
+ * Revised as a template class so it can be used with any supported Adafruit display library,
+ * such as Adafruit_ILI9341, Adafruit_ST7735, etc.
+ * June 2019 mikee47 <mikee@sillyhouse.net>
+ */
+
+#pragma once
+
+#include <SmingCore.h>
```

Comment:
(Nice-to-have) If we use this include then the code won't be easily reusable by the Arduino community. Would it be possible to replace it with <Arduino.h> and other standard Arduino includes?

Reply:
Good point, I've replaced with Arduino.h which is sufficient to compile it. However, it does use the Sming file API so isn't portable from that perspective. There are various solutions to that but my motivation for including this PR now was the issue of maintaining two (almost) identical source code files.